src/CrossValidation/Reports/ResidualAnalysis.php

```diff
@@ -60,6 +60,7 @@ public function generate(array $predictions, array $labels) : array
 
             $l1[] = abs($error);
             $l2[] = $t = $error ** 2;
+            $log[] = (log(1 + $label) - log(1 + $prediction)) ** 2;
```

Comment:
Can this be optimized?

`(log((1 + $label) / (1 + $prediction))) ** 2`

Reply:
At the moment I don't know another solution.
![msle_01](https://user-images.githubusercontent.com/8239917/57704487-49e30c80-7662-11e9-9469-01fe86b55253.png)

Reply:
Since log(x) - log(y) is the same as log(x / y) ... can we arrange the formula such that we only have to compute the log once for each MSLE?

Ex.
```
(log((1 + $label) / (1 + $prediction))) ** 2
```

Reply:
Yep, we can :+1: